metrics/network/remote_network/remote-networking-iperf3.sh

```diff
@@ -100,81 +115,107 @@ function remote_network_parallel_iperf3 {
-	echo "Parallel network is : $total_parallel"
-
+	command="iperf3 -c $server_ip_address -P4 -t $server_time"
+	check_iperf3_client_command "$command"
+	result=$(start_client "$extra_args" "$client_id" "$client_command")
+	total_parallel=$(echo "$result" | tail -n 4 | head -1 | awk '{print $(NF-3)}')
+	units=$(echo "$result" | tail -n 4 | head -1 | awk '{print $(NF-2)}')
+	echo "Parallel network is : $total_parallel $units"
+
+	save_results "$test_name" "Remote parallel" "$total_parallel" "$units"
 	clean_environment
 }
 
 # This function will measure latency across containers
 function remote_network_latency {
+	test_name="Remote network latency"
+	get_runtime
 	#Number of packages
 	number_of_packages=10
+	units="ms"
 	setup_swarm
 	client_replica_status
 	server_replica_status
 
 	server_ip_address=$(check_server_address)
 	client_id=$($DOCKER_EXE ps -q)
 	client_command="ping -c "$number_of_packages" "$server_ip_address""
-	result=$(start_client "$client_id" "$client_command")
+	result=$(start_client "$extra_args" "$client_id" "$client_command")
 	total_latency=$(echo $result | grep avg | awk '{print $(NF-1)}' | cut -d '/' -f 2)
-	echo "Network latency is : $total_latency ms"
+	echo "Network latency is : $total_latency $units"
 
+	save_results "$test_name" "Remote latency" "$total_latency" "$units"
 	clean_environment
 }
 
 function main {
+	[[ $# -ne 7 ]]&& help && die "Illegal number of parameters."
+
 	local OPTIND
-	while getopts "hbjplt:i:u:a" opt
+	while getopts ":a:bhjlpti:u:" opt
 	do
-		case "${opt}" in
+		case "$opt" in
+		a)
+			ssh_address="$OPTARG"
+			;;
+		b)
+			test_bandwidth="1"
+			;;
 		h)
 			help
 			exit 0;
-		;;
-		b)
-			bandwidth_test="1"
-			remote_network_bandwidth_iperf3
-		;;
+                	;;
+		i)
+			interface_name="$OPTARG"
+			;;
 		j)
-			jitter_test="1"
-			remote_network_jitter_iperf3
-		;;
-		p)
-			parallel_test="1"
-			remote_network_parallel_iperf3
-		;;
+			test_jitter="1"
+			;;
 		l)
-			latency_test="1"
-			remote_network_latency
-		;;
+			test_latency="1"
+			;;
+		p)
+			test_parallel="1"
+			;;
 		t)
-			total_test="1"
-			remote_network_bandwidth_iperf3
-			remote_network_jitter_iperf3
-			remote_network_parallel_iperf3
-			remote_network_latency
-		;;
-		i)
-			interface_name="${OPTARG}"
-		;;
+			test_total="1"
+			;;
 		u)
-			ssh_user="${OPTARG}"
-		;;
-		a)
-			ssh_address="${OPTARG}"
-                ;;
+			ssh_user="$OPTARG"
+			;;
+		\?)
+			echo "An invalid option has been entered: -$OPTARG";
+			help
+			exit 0;
+			;;
+		:)
+			echo "Missing argument for -$OPTARG";
+			help
+			exit 0;
+			;;
 		esac
-		shift
 	done
 	shift $((OPTIND-1))
 
-	[ -z "$ssh_address" ] && help && die "Mandatory IP address of host B not supplied"
-	[ -z "$ssh_user" ] && help && die "Mandatory user of host B not supplied"
-	[ -z "$interface_name" ] && help && die "Mandatory interface name to run Swarm not supplied"
-
+	[[ -z "$interface_name" ]] && help && die "Missing IP Address."
```

Comment:
@grahamwhaley and @MarioCarrilloA here it is that validation

Reply:
great, then maybe is unnecessary this `[[ $# -ne 7 ]]&& help && die "Illegal number of parameters."`, and the order of the arguments should be irrelevant for this test execution.

Reply:
@MarioCarrilloA yes the order of the arguments is not relevant